test/extensions/tracers/datadog/tracer_test.cc

```diff
@@ -0,0 +1,197 @@
+#include "envoy/tracing/trace_reason.h"
+
+#include "source/common/tracing/null_span_impl.h"
+#include "source/extensions/tracers/datadog/span.h"
+#include "source/extensions/tracers/datadog/tracer.h"
+
+#include "test/mocks/thread_local/mocks.h"
+#include "test/mocks/upstream/cluster_manager.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "datadog/error.h"
+#include "datadog/expected.h"
+#include "datadog/sampling_priority.h"
+#include "datadog/trace_segment.h"
+#include "datadog/tracer_config.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+namespace {
+
+class DatadogTracerTest : public testing::Test {
+public:
+  DatadogTracerTest() {
+    cluster_manager_.initializeClusters({"fake_cluster"}, {});
+    cluster_manager_.thread_local_cluster_.cluster_.info_->name_ = "fake_cluster";
+    cluster_manager_.initializeThreadLocalClusters({"fake_cluster"});
+  }
+
+protected:
+  NiceMock<Upstream::MockClusterManager> cluster_manager_;
+  Stats::TestUtil::TestStore store_;
+  NiceMock<ThreadLocal::MockInstance> thread_local_slot_allocator_;
+  Event::SimulatedTimeSystem time_;
+};
+
+TEST_F(DatadogTracerTest, Breathing) {
+  // Verify that constructing a `Tracer` instance with mocked dependencies
+  // does not throw exceptions.
+  datadog::tracing::TracerConfig config;
+  config.defaults.service = "envoy";
+
+  Tracer tracer("fake_cluster", "test_host", config, cluster_manager_, *store_.rootScope(),
+                thread_local_slot_allocator_);
+
+  (void)tracer;
```

Comment:
is this needed to avoid a compile error?

Envoy style is to use UNREFERENCED_PARAMETER from ./source/common/common/macros.h